core/domain/translation_services.py

```diff
@@ -56,13 +55,19 @@ def get_and_cache_machine_translation(
     if translation is not None:
         return translation.translated_text
 
-    translated_text = cloud_translate_services.translate_text(
-        source_text, source_language_code, target_language_code)
+    translated_text = None
+    try:
+        translated_text = cloud_translate_services.translate_text(
+            source_text, source_language_code, target_language_code)
+    except ValueError as e: # Language code not allowlisted
```

Comment:
Please put the comment above this line (not at the side), have it end with a period, and provide a bit more detail as a complete sentence. Also explain why we merely log the error instead of letting it raise, so that future developers have context on why the decision was made.

Reply:
Done.